core/unit_test/TestViewMapping_a.hpp

```diff
@@ -653,64 +653,66 @@ void test_view_mapping() {
-    ASSERT_TRUE(
-        (std::is_same_v<typename T::const_scalar_array_type, const int*>));
-    ASSERT_TRUE(
-        (std::is_same_v<typename T::non_const_scalar_array_type, int*>));
+    static_assert(std::is_same_v<typename C::data_type, const int*>);
+    static_assert(std::is_same_v<typename C::const_data_type, const int*>);
+    static_assert(std::is_same_v<typename C::non_const_data_type, int*>);
 
-    ASSERT_TRUE((std::is_same_v<typename T::value_type, int>));
-    ASSERT_TRUE((std::is_same_v<typename T::const_value_type, const int>));
-    ASSERT_TRUE((std::is_same_v<typename T::non_const_value_type, int>));
+    static_assert(std::is_same_v<typename C::scalar_array_type, const int*>);
+    static_assert(
+        std::is_same_v<typename C::const_scalar_array_type, const int*>);
+    static_assert(
+        std::is_same_v<typename C::non_const_scalar_array_type, int*>);
 
-    ASSERT_TRUE((std::is_same_v<typename T::memory_space,
-                                typename Space::memory_space>));
-    ASSERT_TRUE((std::is_same_v<typename T::reference_type, int&>));
+    static_assert(std::is_same_v<typename C::value_type, const int>);
+    static_assert(std::is_same_v<typename C::const_value_type, const int>);
+    static_assert(std::is_same_v<typename C::non_const_value_type, int>);
 
-    ASSERT_EQ(T::rank, size_t(1));
+    static_assert(
+        std::is_same_v<typename C::memory_space, typename Space::memory_space>);
+    static_assert(std::is_same_v<typename C::reference_type, const int&>);
 
-    ASSERT_TRUE((std::is_same_v<typename C::data_type, const int*>));
-    ASSERT_TRUE((std::is_same_v<typename C::const_data_type, const int*>));
-    ASSERT_TRUE((std::is_same_v<typename C::non_const_data_type, int*>));
+    static_assert(C::rank == size_t(1));
 
-    ASSERT_TRUE((std::is_same_v<typename C::scalar_array_type, const int*>));
-    ASSERT_TRUE(
-        (std::is_same_v<typename C::const_scalar_array_type, const int*>));
-    ASSERT_TRUE(
-        (std::is_same_v<typename C::non_const_scalar_array_type, int*>));
+    if (Kokkos::SpaceAccessibility<Kokkos::HostSpace,
+                                   typename Space::memory_space>::accessible &&
+        Kokkos::SpaceAccessibility<typename Space::memory_space,
+                                   Kokkos::HostSpace>::assignable) {
+      int data[N];
 
-    ASSERT_TRUE((std::is_same_v<typename C::value_type, const int>));
-    ASSERT_TRUE((std::is_same_v<typename C::const_value_type, const int>));
-    ASSERT_TRUE((std::is_same_v<typename C::non_const_value_type, int>));
+      T vr1(data, N);  // View of non-const.
+      C cr1(vr1);      // View of const from view of non-const.
+      C cr2((const int*)data, N);
 
-    ASSERT_TRUE((std::is_same_v<typename C::memory_space,
-                                typename Space::memory_space>));
-    ASSERT_TRUE((std::is_same_v<typename C::reference_type, const int&>));
+      // Generate static_assert error:
+      // T tmp( cr1 );
```

Comment:
What was this for?

Reply:
I think it's just a reminder that this case was considered but, of course, doesn't compile. I simple left it.